src/CBT.Core/Internal/DefaultNuGetDownloader.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Microsoft.Build.Framework;
+using System;
+using System.IO;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CBT.Core.Internal
+{
+    /// <summary>
+    /// A class that will download the latest version of NuGet from NuGet.org.
+    /// </summary>
+    internal static class DefaultNuGetDownloader
+    {
+        private static readonly Uri NuGetUrl = new Uri("https://dist.nuget.org/win-x86-commandline/latest/nuget.exe");
```

Comment:
the other potential use case is to down load another package tool which isn't nuget because at some point we said hey lets host cbt modules as xxx packages or some other consumers of cbt have a hard no nuget requirement...

either way it is fine as it is if there becomes a real use case we can refractor.

On May 19, 2016 8:20 PM, josemesona <notifications@github.com> wrote:

In src/CBT.Core/Internal/DefaultNuGetDownloader.cs<https://github.com/CommonBuildToolset/CBT.Core/pull/9#discussion_r63986525>:

> @@ -0,0 +1,83 @@
> +﻿using Microsoft.Build.Framework;
> +using System;
> +using System.IO;
> +using System.Net;
> +using System.Threading;
> +using System.Threading.Tasks;
> +
> +namespace CBT.Core.Internal
> +{
> +    /// <summary>
> +    /// A class that will download the latest version of NuGet from NuGet.org.
> +    /// </summary>
> +    internal static class DefaultNuGetDownloader
> +    {
> +        private static readonly Uri NuGetUrl = new Uri("https://dist.nuget.org/win-x86-commandline/latest/nuget.exe");


I thought about this but the only other use case is to download the credential provider bundle which is a zip file and not multiple files. I'll send out a pull request tomorrow for the implementation of this which downloads the credential provider bundle and unzips it.

—
You are receiving this because you were assigned.
Reply to this email directly or view it on GitHub<https://github.com/CommonBuildToolset/CBT.Core/pull/9/files/818a836b36efe6530c86346c59c37f40ac1cd822#r63986525>